internal/worker/uniter/runner/context/context.go

```diff
@@ -1250,6 +1252,55 @@ func (ctx *HookContext) CloudSpec() (*params.CloudSpec, error) {
 	return ctx.cloudSpec, nil
 }
 
+// cloudSpecK8s loads the in-cluster configuration to connect to the Kubernetes API.
+//
+// Loosely based on the Kubernetes Go client's rest.InClusterConfig function:
+// https://github.com/kubernetes/client-go/blob/0341f077c9d600b7a68d6f637cbd5da9ace4df3d/rest/config.go#L543
```

Comment:
Why not just call `rest.InClusterConfig`?

Reply:
Done. Still need to add tests, so no need to review just yet.